app/react/Pages/utils/specs/setPageAssets.spec.ts

```diff
@@ -0,0 +1,139 @@
+    expect(itemLists.length).toBe(4);
+    expect(itemLists[0].params).toBe('?q=(a:1,b:2)');
+    expect(itemLists[0].items).toEqual(['resultsFor:0']);
+    expect(itemLists[1].params).toBe('');
+    expect(itemLists[1].items).toEqual(['resultsFor:1']);
+    expect(itemLists[2].params).toBe('?q=(x:1,y:!(%27array%27),limit:24)');
+    expect(itemLists[2].items).toEqual(['resultsFor:2']);
+    expect(itemLists[3].params).toBe('?order=metadata.form&treatAs=number');
+    expect(itemLists[3].items).toEqual(['resultsFor:3']);
+    expect(itemLists[3].options).toEqual({ limit: 12 });
+  };
+
+  it('should request each list inside the content limited to 6 items (default) or the passed value and set the state', async () => {
+    const stateActions = await setPageAssets(request);
+
+    expect(apiSearch.calls.count()).toBe(4);
+    expect(JSON.parse(JSON.stringify(apiSearch.calls.argsFor(0)[0]))).toEqual({
+      data: { a: 1, b: 2, limit: '6' },
+      headers: 'headers',
+    });
+    expect(apiSearch.calls.argsFor(1)[0]).toEqual({
+      data: { filters: {}, types: [], limit: '9' },
+      headers: 'headers',
+    });
+
+    expect(JSON.parse(JSON.stringify(apiSearch.calls.argsFor(2)[0]))).toEqual({
+      data: {
+        x: 1,
+        y: ['array'],
+        limit: '6',
+      },
+      headers: 'headers',
+    });
+    expect(apiSearch.calls.argsFor(3)[0]).toEqual({
+      data: { filters: {}, types: [], limit: '12' },
+      headers: 'headers',
+    });
+
+    const itemLists = stateActions[1].value;
+    assertItemLists(itemLists);
+  });
+
+  describe('Datasets', () => {
+    let markdownDatasetsResponse: {};
+    const getDatasetsActions = (stateActions: Action[]) =>
+      stateActions.find(a => a.type === 'page/datasets/SET')?.value;
+
+    beforeEach(() => {
+      markdownDatasetsResponse = { request1: 'url1', request2: 'url2' };
+      spyOn(markdownDatasets, 'fetch').and.callFake(async (content, requestParams, options) => {
+        expect(content).toBe('originalContent');
+        expect(requestParams).toEqual(request.onlyHeaders());
+        return Promise.resolve({ ...markdownDatasetsResponse, ...options.additionalDatasets });
+      });
+    });
+
+    it('should request each dataset inside the content', async () => {
+      const stateActions = await setPageAssets(request);
+      expect(getDatasetsActions(stateActions)).toEqual(markdownDatasetsResponse);
+    });
+
+    describe('Extended datasets and data', () => {
+      it('should request additional dataset queries and passed data', async () => {
+        const stateActions = await setPageAssets(
+          request,
+          { customDataset: { url: 'someurl' } },
+          { customData: { localData: 'from store' } }
+        );
+
+        expect(getDatasetsActions(stateActions)).toEqual({
+          request1: 'url1',
+          request2: 'url2',
+          customDataset: { url: 'someurl' },
+          customData: { localData: 'from store' },
+        });
```

Comment:
this is easier to test if the setPagesAssets only returns the values and not the actions.

Reply:
Maybe, but then we would still need to abstract the setting of the actions so they can be called from two different routes, so the test of this function will be simpler, but the test of the "parent" function will still look like this.  Do you think this merits the change still?

Reply:
the initial reason was for better concern separation and readability of other functions, so yes i think it merits.